drivers/spi/spi_andes_atcspi200.c

```diff
@@ -162,30 +190,386 @@ static int configure(const struct device *dev,
 	return 0;
 }
 
-static void transfer_next_chunk(const struct device *dev)
+
+#ifdef CONFIG_ANDES_SPI_DMA_MODE
+
+static int spi_dma_tx_load(const struct device *dev);
+static int spi_dma_rx_load(const struct device *dev);
+
+static inline void spi_tx_dma_enable(const struct device *dev)
+{
+	/* Enable TX DMA */
+	SET_MASK(SPI_CTRL(dev), CTRL_TX_DMA_EN_MSK);
+}
+
+static inline void spi_tx_dma_disable(const struct device *dev)
+{
+	/* Disable TX DMA */
+	CLR_MASK(SPI_CTRL(dev), CTRL_TX_DMA_EN_MSK);
+}
+
+static inline void spi_rx_dma_enable(const struct device *dev)
+{
+	/* Enable RX DMA */
+	SET_MASK(SPI_CTRL(dev), CTRL_RX_DMA_EN_MSK);
+}
+
+static inline void spi_rx_dma_disable(const struct device *dev)
+{
+	/* Disable RX DMA */
+	CLR_MASK(SPI_CTRL(dev), CTRL_RX_DMA_EN_MSK);
+}
+
+static int spi_dma_move_buffers(const struct device *dev)
+{
+	struct spi_atcspi200_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	uint32_t error = 0;
+
+	data->dma_rx.dma_blk_cfg.next_block = NULL;
+	data->dma_tx.dma_blk_cfg.next_block = NULL;
+
+	if (spi_context_tx_on(ctx)) {
+		error = spi_dma_tx_load(dev);
+		if (error != 0) {
+			return error;
+		}
+	}
+
+	if (spi_context_rx_on(ctx)) {
+		error = spi_dma_rx_load(dev);
+		if (error != 0) {
+			return error;
+		}
+	}
+
+	return 0;
+}
+
+static inline void dma_rx_callback(const struct device *dev, void *user_data,
+				uint32_t channel, int status)
+{
+	struct spi_atcspi200_data *data = user_data;
+	const struct device *spi_dev = data->self_dev;
+	struct spi_context *ctx = &data->ctx;
+
+	dma_stop(data->dma_rx.dma_dev, data->dma_rx.channel);
+	spi_rx_dma_disable(spi_dev);
+
+	if (spi_context_rx_on(ctx)) {
+		spi_dma_rx_load(spi_dev);
```

Comment:
the error return isn't caught here, (void)spi_dma_rx_load(spi_dev) to signal the return is ignored perhaps